src/Aspire.Dashboard/Model/ResourceSourceViewModel.cs

```diff
@@ -1,47 +1,79 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Globalization;
+using System.Text.RegularExpressions;
+
 namespace Aspire.Dashboard.Model;
 
-public class ResourceSourceViewModel(string value, string? contentAfterValue, string valueToVisualize, string tooltip)
+public class ResourceSourceViewModel(string value, List<LaunchArgument>? contentAfterValue, string valueToVisualize, string tooltip)
 {
     public string Value { get; } = value;
-    public string? ContentAfterValue { get; } = contentAfterValue;
+    public List<LaunchArgument>? ContentAfterValue { get; } = contentAfterValue;
     public string ValueToVisualize { get; } = valueToVisualize;
     public string Tooltip { get; } = tooltip;
 
     internal static ResourceSourceViewModel? GetSourceViewModel(ResourceViewModel resource)
     {
-        var executablePath = resource.TryGetExecutablePath(out var path) ? path : null;
-
-        (string? ArgumentsString, string FullCommandLine)? commandLineInfo = null;
+        (List<LaunchArgument>? Arguments, string? ArgumentsString)? commandLineInfo;
 
-        if (resource.TryGetExecutableArguments(out var arguments))
+        // If the resource contains launch arguments, these project arguments should be shown in place of all executable arguments,
+        // which include args added by the app host
+        if (resource.TryGetAppArgs(out var launchArguments))
         {
-            var argumentsString = arguments.IsDefaultOrEmpty ? null : string.Join(" ", arguments);
-            commandLineInfo = (ArgumentsString: argumentsString, $"{executablePath} {argumentsString}");
+            if (launchArguments.IsDefaultOrEmpty)
+            {
+                commandLineInfo = (null, null);
+            }
+            else
+            {
+                var argumentsString = string.Join(" ", launchArguments);
+                if (resource.TryGetAppArgsFormatParams(out var formatParams))
+                {
+                    var arguments = launchArguments
+                        .Select(arg => new LaunchArgument(arg, !Regex.IsMatch(arg, "{\\d+}")))
```

Comment:
So I know this is not super likely to happen... but what if the program invocation parameter needs to be, literally, something like `{12}`? Isn't this going to be a problem here?

What if we did the `(sensitive value) --> placeholder` substitution before we write the annotation onto the DCP object? Then the view model could read the annotation directly with minimal processing.

Reply:
If we need both the "masked" as well as "fully resolved" arguments (depending on the view selected by the user in the dashboard), we could store two versions of the argument list/string in the annotation.

Reply:
> what if the program invocation parameter needs to be, literally, something like {12}? Isn't this going to be a problem here

Yeah, that's a good point. I decided instead to avoid the format string entirely, and use an array of booleans representing sensitivity (seemingly protobuf won't serialize an immutablearray<bool>, so it's immutablearray<int>)

`resource.appArgs`: `one two three SECRET five`
`resource.appArgsSensitivity` `[false, false, false, true, false]`

> What if we did the (sensitive value) --> placeholder substitution before we write the annotation onto the DCP object? Then the view model could read the annotation directly with minimal processing.

With the change above, the substitution is no longer required :)

Reply:
Nice